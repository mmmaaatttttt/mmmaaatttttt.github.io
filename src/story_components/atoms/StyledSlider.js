import React, { Component } from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import Slider from "rc-slider";
import { darken } from "polished";

const darkerMain = props => darken(0.2, props.mainColor);
const totalHeight = s => props => (props.height + 2 * props.padding) * s;

const StyledSlider = styled(Slider)`
  position: relative;
  height: ${totalHeight(1)}px;
  padding: ${props => props.padding}px 0;
  width: 100%;
  border-radius: ${totalHeight(1)}px;
  -ms-touch-action: none;
  touch-action: none;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  * {
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  .rc-slider-rail {
    position: absolute;
    width: 100%;
    background-color: #e9e9e9;
    height: ${props => props.height}px;
    border-radius: ${totalHeight(1)}px;
  }

  .rc-slider-track {
    position: absolute;
    left: 0;
    height: ${props => props.height}px;
    border-radius: ${totalHeight(1)}px;
    background-color: ${props => props.mainColor};
  }

  .rc-slider-handle {
    position: absolute;
    margin-left: -${totalHeight(1 / 2)}px;
    margin-top: -${props => props.padding}px;
    width: ${totalHeight(1)}px;
    height: ${totalHeight(1)}px;
    cursor: pointer;
    cursor: grab;
    border-radius: 50%;
    border: solid ${props => props.padding * 0.75}px ${props => props.mainColor};
    background-color: ${darkerMain};
    touch-action: pan-x;
    transition: border-width 0.5s;
  }

  .rc-slider-handle:hover {
    border-width: 0;
  }

  .rc-slider-handle:active {
    border-width: 0;
    border-color: ${darkerMain};
    box-shadow: 0 0 ${props => props.padding}px ${darkerMain};
    cursor: grabbing;
  }

  .rc-slider-handle:focus {
    border-width: 0;
    border-color: ${darkerMain};
    box-shadow: 0 0 0 ${props => props.padding}px ${darkerMain};
    outline: none;
  }

  .rc-slider-mark {
    position: absolute;
    top: 18px;
    left: 0;
    width: 100%;
    font-size: 12px;
  }

  .rc-slider-mark-text {
    position: absolute;
    display: inline-block;
    vertical-align: middle;
    text-align: center;
    cursor: pointer;
    color: #999;
  }

  .rc-slider-mark-text-active {
    color: #666;
  }

  .rc-slider-step {
    position: absolute;
    width: 100%;
    height: ${props => props.height}px;
    background: transparent;
  }
`;

StyledSlider.propTypes = {
  mainColor: PropTypes.string.isRequired,
  height: PropTypes.number.isRequired,
  padding: PropTypes.number.isRequired
};

StyledSlider.defaultProps = {
  mainColor: "#abe2fb",
  height: 6,
  padding: 10
};

export default StyledSlider;
