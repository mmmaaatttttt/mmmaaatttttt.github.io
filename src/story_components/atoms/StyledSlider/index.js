import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import Slider from "rc-slider";
import { darken } from "polished";

const darkerMain = props => darken(0.2, props.activeColor);
const totalHeight = s => props => (props.height + 2 * props.padding) * s;

const StyledSlider = styled(Slider)`
  position: relative;
  height: ${totalHeight(1)}px;
  padding: ${props => props.padding}px 0;
  width: 100%;
  border-radius: ${totalHeight(1)}px;
  -ms-touch-action: none;
  touch-action: none;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  display: flex;
  justify-content: space-between;

  * {
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  .rc-slider-rail {
    position: absolute;
    width: 100%;
    background-color: ${props => props.inactiveColor};
    height: ${props => props.height}px;
    border-radius: ${totalHeight(1)}px;
  }

  .rc-slider-track {
    position: absolute;
    left: 0;
    height: ${props => props.height}px;
    border-radius: ${totalHeight(1)}px;
    background-color: ${props => props.activeColor};
  }

  .rc-slider-handle {
    position: absolute;
    margin-left: -${totalHeight(1 / 2)}px;
    margin-top: -${props => props.padding}px;
    width: ${totalHeight(1)}px;
    height: ${totalHeight(1)}px;
    cursor: pointer;
    cursor: grab;
    border-radius: 50%;
    border: solid ${props => props.padding * 0.75}px
      ${props => props.activeColor};
    background-color: ${darkerMain};
    touch-action: pan-x;
    transition: border-width 0.5s;
    z-index: 100;
  }

  .rc-slider-handle:hover {
    border-width: 0;
  }

  .rc-slider-handle:active {
    border-width: 0;
    border-color: ${darkerMain};
    box-shadow: 0 0 ${props => props.padding}px ${darkerMain};
    cursor: grabbing;
  }

  .rc-slider-handle:focus {
    border-width: 0;
    border-color: ${darkerMain};
    box-shadow: 0 0 0 ${props => props.padding}px ${darkerMain};
    outline: none;
  }
`;

StyledSlider.propTypes = {
  activeColor: PropTypes.string.isRequired,
  inactiveColor: PropTypes.string.isRequired
};

StyledSlider.defaultProps = {
  activeColor: "#abe2fb",
  inactiveColor: "#e9e9e9"
};

export default StyledSlider;
