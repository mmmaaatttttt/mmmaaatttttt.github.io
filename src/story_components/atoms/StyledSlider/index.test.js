import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import StyledSlider from ".";

const wrap = (props = {}) => shallow(<StyledSlider {...props} />);

it("renders successfully", () => {
  const wrapper = wrap();
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders props when passed in", () => {
  const wrapper = wrap({ activeColor: "red", inactiveColor: "black" });
  expect(toJson(wrapper)).toMatchSnapshot();
});
