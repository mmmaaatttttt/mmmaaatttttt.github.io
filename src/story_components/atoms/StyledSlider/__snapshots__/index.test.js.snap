// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders props when passed in 1`] = `
<ComponentEnhancer(undefined)
  activeColor="red"
  activeDotStyle={Object {}}
  className="sc-bdVaJa iyEPrm"
  disabled={false}
  dotStyle={Object {}}
  dots={false}
  handle={[Function]}
  handleStyle={
    Array [
      Object {},
    ]
  }
  inactiveColor="black"
  included={true}
  marks={Object {}}
  max={100}
  min={0}
  onAfterChange={[Function]}
  onBeforeChange={[Function]}
  onChange={[Function]}
  prefixCls="rc-slider"
  railStyle={Object {}}
  step={1}
  trackStyle={
    Array [
      Object {},
    ]
  }
  vertical={false}
/>
`;

exports[`renders successfully 1`] = `
<ComponentEnhancer(undefined)
  activeColor="#abe2fb"
  activeDotStyle={Object {}}
  className="sc-bdVaJa kbRwvP"
  disabled={false}
  dotStyle={Object {}}
  dots={false}
  handle={[Function]}
  handleStyle={
    Array [
      Object {},
    ]
  }
  inactiveColor="#e9e9e9"
  included={true}
  marks={Object {}}
  max={100}
  min={0}
  onAfterChange={[Function]}
  onBeforeChange={[Function]}
  onChange={[Function]}
  prefixCls="rc-slider"
  railStyle={Object {}}
  step={1}
  trackStyle={
    Array [
      Object {},
    ]
  }
  vertical={false}
/>
`;
